tests/query_tests.rs

```diff
@@ -25,24 +27,38 @@ fn check_compilation(
     // This is the binary to compile a query
     let mut cmd = Command::new(proj_dir.join("target/debug/dtc"));
     // Assemble the args, first the input query
-    let mut args = vec!["-q".to_string(), query_file.to_str().unwrap().into()];
+    let mut args = vec!["-q", query_file.to_str().unwrap()];
     // Append every udf that is provided
+    let mut str_builder = String::new();
     for udf_name in udf_names {
         let udf_file = udf_dir.join(udf_name);
-        args.push("-u".to_string());
-        args.push(udf_file.to_str().unwrap().into());
+        str_builder.push_str("-u");
+        str_builder.push_str(udf_file.to_str().unwrap());
+    }
+    if !str_builder.is_empty() {
+        args.push(&str_builder);
     }
     let mut out_file = query_dir.join(query_name);
     out_file.set_extension("cc");
-    args.extend(vec!["-o".to_string(), out_file.to_str().unwrap().into()]);
-    args.extend(vec![
-        "--root-node".to_string(),
-        "productpage-v1".to_string(),
-    ]);
+    args.extend(vec!["-o", out_file.to_str().unwrap()]);
+    args.extend(vec!["--root-node", "productpage-v1"]);
     cmd.args(args);
     cmd.assert().success();
 
-    Ok(())
+    let mut ref_file = query_dir.join(query_name);
+    ref_file.set_extension("cc.ref");
+    let out_file_str = fs::read_to_string(out_file).unwrap();
+    let ref_file_str = fs::read_to_string(ref_file).unwrap();
+    if out_file_str != ref_file_str {
+        let diff = diffy::create_patch(&out_file_str, &ref_file_str);
+        let diff_color = diffy::PatchFormatter::new().with_color();
+        panic!(
```

Comment:
Why should we panic if they're different?  As we iterate, we change the content in the filters, so it's expected that it changes sometimes.

Reply:
Yes. So when a new compiler change is introduced you have to update the reference files to make sure they reflect the current state of the compiler. Those diffs also help reviewers (e.g., me) understand changes to the code generation. And you can use these reference files as a history of compiler changes to understand when a change may have been introduced. A similar thing are golden files for API tests.

Reply:
Okay, that makes sense.